storage/cloudspanner/log_storage.go

```diff
@@ -141,16 +141,20 @@ func (ls *logStorage) begin(ctx context.Context, tree *trillian.Tree, readonly b
 		return nil, err
 	}
 
-	// Sanity check tx.config
-	if cfg, ok := tx.config.(*spannerpb.LogStorageConfig); !ok || cfg == nil {
-		return nil, fmt.Errorf("unexpected config type for LOG tree %v: %T", tx.treeID, tx.config)
-	}
-
-	return &logTX{
+	ltx := &logTX{
 		ls:       ls,
 		dequeued: make(map[string]*QueuedEntry),
 		treeTX:   tx,
-	}, nil
+	}
+
+	// Needed to generate ErrTreeNeedsInit in SnapshotForTree and other methods.
+	if err := ltx.getLatestRoot(ctx); err == storage.ErrTreeNeedsInit {
```

Comment:
This might be forcing an extra DB scan for an STH when one wouldn't have been done previously (e.g. if the LogServer was requesting just an inclusion proof for a given leaf, perhaps?)

Reply:
It's wasteful, yes, but our current integration tests enforce this property, and I don't know how much of the tower of logic on top relies on this. 
It should be a separate action item to change the API contract of the storage layer.

Reply:
The mysql impl also has this wasteful operation.